pkg/api/util/ipc_endpoint.go

```diff
@@ -0,0 +1,144 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package util
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/util/system"
+)
+
+// IPCEndpoint is an endpoint that IPC requests will be sent to
+type IPCEndpoint struct {
+	client    *http.Client
+	target    url.URL
+	closeConn bool
+}
+
+// GetOption is an option that can be passed to DoGet
+type GetOption func(url.URL) url.URL
+
+// WithValues is an option to add url.Values to the GET request
+func WithValues(values url.Values) GetOption {
+	return func(u url.URL) url.URL {
+		u.RawQuery = values.Encode()
+		return u
+	}
+}
+
+// DoGet sends GET method to the endpoint
+func (end *IPCEndpoint) DoGet(options ...GetOption) ([]byte, error) {
+	conn := LeaveConnectionOpen
+	if end.closeConn {
+		conn = CloseConnection
+	}
+
+	target := end.target
+	for _, opt := range options {
+		target = opt(target)
+	}
+
+	// TODO: after removing callers to api/util/DoGet, pass `end.token` instead of using global var
+	res, err := DoGet(end.client, target.String(), conn)
+	if err != nil {
+		var errMap = make(map[string]string)
+		_ = json.Unmarshal(res, &errMap) //nolint:errcheck
+		// If the error has been marshalled into a json object, check it and return it properly
+		if errStr, found := errMap["error"]; found {
+			return nil, errors.New(errStr)
+		}
+
+		return nil, fmt.Errorf("could not reach agent: %v\nMake sure the agent is running before requesting the runtime configuration and contact support if you continue having issues", err)
+	}
+	return res, err
+}
+
+// EndpointOption allows configuration of the IPCEndpoint during construction
+type EndpointOption func(*IPCEndpoint)
+
+// WithCloseConnection is an option to close the connection
+func WithCloseConnection(state bool) func(*IPCEndpoint) {
+	return func(end *IPCEndpoint) {
+		end.closeConn = state
+	}
+}
+
+// WithHTTPClient is an option to assign a different http.Client
+func WithHTTPClient(client *http.Client) func(*IPCEndpoint) {
+	return func(end *IPCEndpoint) {
+		end.client = client
+	}
+}
+
+// WithURLScheme is an option to set the URL's scheme
+func WithURLScheme(scheme string) func(*IPCEndpoint) {
+	return func(end *IPCEndpoint) {
+		end.target.Scheme = scheme
+	}
+}
+
+// WithHostAndPort is an option to use a host address for sending IPC requests
+func WithHostAndPort(cmdHost string, cmdPort int) func(*IPCEndpoint) {
+	return func(end *IPCEndpoint) {
+		end.target.Host = net.JoinHostPort(cmdHost, strconv.Itoa(cmdPort))
+	}
+}
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏I think it would be nice specifying what the default value of each of those is (otherwise I guess you need to look in the code to know if you actually need to use the option)